Public/Final/tor-0.3.3.9/usr/local/share/man/man1/tor.1
'\" t
.\"     Title: tor
.\"    Author: [see the "AUTHORS" section]
.\" Generator: DocBook XSL Stylesheets v1.78.1 <http://docbook.sf.net/>
.\"      Date: 07/13/2018
.\"    Manual: Tor Manual
.\"    Source: Tor
.\"  Language: English
.\"
.TH "TOR" "1" "07/13/2018" "Tor" "Tor Manual"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
tor \- The second\-generation onion router
.SH "SYNOPSIS"
.sp
\fBtor\fR [\fIOPTION\fR \fIvalue\fR]\&...
.SH "DESCRIPTION"
.sp
Tor is a connection\-oriented anonymizing communication service\&. Users choose a source\-routed path through a set of nodes, and negotiate a "virtual circuit" through the network, in which each node knows its predecessor and successor, but no others\&. Traffic flowing down the circuit is unwrapped by a symmetric key at each node, which reveals the downstream node\&.
.sp
Basically, Tor provides a distributed network of servers or relays ("onion routers")\&. Users bounce their TCP streams \(em web traffic, ftp, ssh, etc\&. \(em around the network, and recipients, observers, and even the relays themselves have difficulty tracking the source of the stream\&.
.sp
By default, \fBtor\fR will act as a client only\&. To help the network by providing bandwidth as a relay, change the \fBORPort\fR configuration option \(em see below\&. Please also consult the documentation on the Tor Project\(cqs website\&.
.SH "COMMAND-LINE OPTIONS"
.PP
\fB\-h\fR, \fB\-help\fR
.RS 4
Display a short help message and exit\&.
.RE
.PP
\fB\-f\fR \fIFILE\fR
.RS 4
Specify a new configuration file to contain further Tor configuration options OR pass
\fB\-\fR
to make Tor read its configuration from standard input\&. (Default: /etc/tor/torrc, or $HOME/\&.torrc if that file is not found)
.RE
.PP
\fB\-\-allow\-missing\-torrc\fR
.RS 4
Do not require that configuration file specified by
\fB\-f\fR
exist if default torrc can be accessed\&.
.RE
.PP
\fB\-\-defaults\-torrc\fR \fIFILE\fR
.RS 4
Specify a file in which to find default values for Tor options\&. The contents of this file are overridden by those in the regular configuration file, and by those on the command line\&. (Default: /etc/tor/torrc\-defaults\&.)
.RE
.PP
\fB\-\-ignore\-missing\-torrc\fR
.RS 4
Specifies that Tor should treat a missing torrc file as though it were empty\&. Ordinarily, Tor does this for missing default torrc files, but not for those specified on the command line\&.
.RE
.PP
\fB\-\-hash\-password\fR \fIPASSWORD\fR
.RS 4
Generates a hashed password for control port access\&.
.RE
.PP
\fB\-\-list\-fingerprint\fR
.RS 4
Generate your keys and output your nickname and fingerprint\&.
.RE
.PP
\fB\-\-verify\-config\fR
.RS 4
Verify the configuration file is valid\&.
.RE
.PP
\fB\-\-service install\fR [\fB\-\-options\fR \fIcommand\-line options\fR]
.RS 4
Install an instance of Tor as a Windows service, with the provided command\-line options\&. Current instructions can be found at
https://www\&.torproject\&.org/docs/faq#NTService
.RE
.PP
\fB\-\-service\fR \fBremove\fR|\fBstart\fR|\fBstop\fR
.RS 4
Remove, start, or stop a configured Tor Windows service\&.
.RE
.PP
\fB\-\-nt\-service\fR
.RS 4
Used internally to implement a Windows service\&.
.RE
.PP
\fB\-\-list\-torrc\-options\fR
.RS 4
List all valid options\&.
.RE
.PP
\fB\-\-list\-deprecated\-options\fR
.RS 4
List all valid options that are scheduled to become obsolete in a future version\&. (This is a warning, not a promise\&.)
.RE
.PP
\fB\-\-version\fR
.RS 4
Display Tor version and exit\&.
.RE
.PP
\fB\-\-quiet\fR|\fB\-\-hush\fR
.RS 4
Override the default console log\&. By default, Tor starts out logging messages at level "notice" and higher to the console\&. It stops doing so after it parses its configuration, if the configuration tells it to log anywhere else\&. You can override this behavior with the
\fB\-\-hush\fR
option, which tells Tor to only send warnings and errors to the console, or with the
\fB\-\-quiet\fR
option, which tells Tor not to log to the console at all\&.
.RE
.PP
\fB\-\-keygen\fR [\fB\-\-newpass\fR]
.RS 4
Running "tor \-\-keygen" creates a new ed25519 master identity key for a relay, or only a fresh temporary signing key and certificate, if you already have a master key\&. Optionally you can encrypt the master identity key with a passphrase: Tor will ask you for one\&. If you don\(cqt want to encrypt the master key, just don\(cqt enter any passphrase when asked\&.

The
\fB\-\-newpass\fR
option should be used with \-\-keygen only when you need to add, change, or remove a passphrase on an existing ed25519 master identity key\&. You will be prompted for the old passphase (if any), and the new passphrase (if any)\&.

When generating a master key, you will probably want to use
\fB\-\-DataDirectory\fR
to control where the keys and certificates will be stored, and
\fB\-\-SigningKeyLifetime\fR
to control their lifetimes\&. Their behavior is as documented in the server options section below\&. (You must have write access to the specified DataDirectory\&.)

To use the generated files, you must copy them to the DataDirectory/keys directory of your Tor daemon, and make sure that they are owned by the user actually running the Tor daemon on your system\&.
.RE
.PP
\fB\-\-passphrase\-fd\fR \fIFILEDES\fR
.RS 4
Filedescriptor to read the passphrase from\&. Note that unlike with the tor\-gencert program, the entire file contents are read and used as the passphrase, including any trailing newlines\&. Default: read from the terminal\&.
.RE
.PP
\fB\-\-key\-expiration\fR [\fBpurpose\fR]
.RS 4
The
\fBpurpose\fR
specifies which type of key certificate to determine the expiration of\&. The only currently recognised
\fBpurpose\fR
is "sign"\&.

Running "tor \-\-key\-expiration sign" will attempt to find your signing key certificate and will output, both in the logs as well as to stdout, the signing key certificate\(cqs expiration time in ISO\-8601 format\&. For example, the output sent to stdout will be of the form: "signing\-cert\-expiry: 2017\-07\-25 08:30:15 UTC"
.RE
.sp
Other options can be specified on the command\-line in the format "\-\-option value", in the format "option value", or in a configuration file\&. For instance, you can tell Tor to start listening for SOCKS connections on port 9999 by passing \-\-SocksPort 9999 or SocksPort 9999 to it on the command line, or by putting "SocksPort 9999" in the configuration file\&. You will need to quote options with spaces in them: if you want Tor to log all debugging messages to debug\&.log, you will probably need to say \-\-Log \fIdebug file debug\&.log\fR\&.
.sp
Options on the command line override those in configuration files\&. See the next section for more information\&.
.SH "THE CONFIGURATION FILE FORMAT"
.sp
All configuration options in a configuration are written on a single line by default\&. They take the form of an option name and a value, or an option name and a quoted value (option value or option "value")\&. Anything after a # character is treated as a comment\&. Options are case\-insensitive\&. C\-style escaped characters are allowed inside quoted values\&. To split one configuration entry into multiple lines, use a single backslash character (\e) before the end of the line\&. Comments can be used in such multiline entries, but they must start at the beginning of a line\&.
.sp
Configuration options can be imported from files or folders using the %include option with the value being a path\&. If the path is a file, the options from the file will be parsed as if they were written where the %include option is\&. If the path is a folder, all files on that folder will be parsed following lexical order\&. Files starting with a dot are ignored\&. Files on subfolders are ignored\&. The %include option can be used recursively\&.
.sp
By default, an option on the command line overrides an option found in the configuration file, and an option in a configuration file overrides one in the defaults file\&.
.sp
This rule is simple for options that take a single value, but it can become complicated for options that are allowed to occur more than once: if you specify four SocksPorts in your configuration file, and one more SocksPort on the command line, the option on the command line will replace \fIall\fR of the SocksPorts in the configuration file\&. If this isn\(cqt what you want, prefix the option name with a plus sign (+), and it will be appended to the previous set of options instead\&. For example, setting SocksPort 9100 will use only port 9100, but setting +SocksPort 9100 will use ports 9100 and 9050 (because this is the default)\&.
.sp
Alternatively, you might want to remove every instance of an option in the configuration file, and not replace it at all: you might want to say on the command line that you want no SocksPorts at all\&. To do that, prefix the option name with a forward slash (/)\&. You can use the plus sign (+) and the forward slash (/) in the configuration file and on the command line\&.
.SH "GENERAL OPTIONS"
.PP
\fBBandwidthRate\fR \fIN\fR \fBbytes\fR|\fBKBytes\fR|\fBMBytes\fR|\fBGBytes\fR|\fBTBytes\fR|\fBKBits\fR|\fBMBits\fR|\fBGBits\fR|\fBTBits\fR
.RS 4
A token bucket limits the average incoming bandwidth usage on this node to the specified number of bytes per second, and the average outgoing bandwidth usage to that same value\&. If you want to run a relay in the public network, this needs to be
\fIat the very least\fR
75 KBytes for a relay (that is, 600 kbits) or 50 KBytes for a bridge (400 kbits) \(em but of course, more is better; we recommend at least 250 KBytes (2 mbits) if possible\&. (Default: 1 GByte)

Note that this option, and other bandwidth\-limiting options, apply to TCP data only: They do not count TCP headers or DNS traffic\&.

With this option, and in other options that take arguments in bytes, KBytes, and so on, other formats are also supported\&. Notably, "KBytes" can also be written as "kilobytes" or "kb"; "MBytes" can be written as "megabytes" or "MB"; "kbits" can be written as "kilobits"; and so forth\&. Tor also accepts "byte" and "bit" in the singular\&. The prefixes "tera" and "T" are also recognized\&. If no units are given, we default to bytes\&. To avoid confusion, we recommend writing "bytes" or "bits" explicitly, since it\(cqs easy to forget that "B" means bytes, not bits\&.
.RE
.PP
\fBBandwidthBurst\fR \fIN\fR \fBbytes\fR|\fBKBytes\fR|\fBMBytes\fR|\fBGBytes\fR|\fBTBytes\fR|\fBKBits\fR|\fBMBits\fR|\fBGBits\fR|\fBTBits\fR
.RS 4
Limit the maximum token bucket size (also known as the burst) to the given number of bytes in each direction\&. (Default: 1 GByte)
.RE
.PP
\fBMaxAdvertisedBandwidth\fR \fIN\fR \fBbytes\fR|\fBKBytes\fR|\fBMBytes\fR|\fBGBytes\fR|\fBTBytes\fR|\fBKBits\fR|\fBMBits\fR|\fBGBits\fR|\fBTBits\fR
.RS 4
If set, we will not advertise more than this amount of bandwidth for our BandwidthRate\&. Server operators who want to reduce the number of clients who ask to build circuits through them (since this is proportional to advertised bandwidth rate) can thus reduce the CPU demands on their server without impacting network performance\&.
.RE
.PP
\fBRelayBandwidthRate\fR \fIN\fR \fBbytes\fR|\fBKBytes\fR|\fBMBytes\fR|\fBGBytes\fR|\fBTBytes\fR|\fBKBits\fR|\fBMBits\fR|\fBGBits\fR|\fBTBits\fR
.RS 4
If not 0, a separate token bucket limits the average incoming bandwidth usage for _relayed traffic_ on this node to the specified number of bytes per second, and the average outgoing bandwidth usage to that same value\&. Relayed traffic currently is calculated to include answers to directory requests, but that may change in future versions\&. They do not include directory fetches by the relay (from authority or other relays), because that is considered "client" activity\&. (Default: 0)
.RE
.PP
\fBRelayBandwidthBurst\fR \fIN\fR \fBbytes\fR|\fBKBytes\fR|\fBMBytes\fR|\fBGBytes\fR|\fBTBytes\fR|\fBKBits\fR|\fBMBits\fR|\fBGBits\fR|\fBTBits\fR
.RS 4
If not 0, limit the maximum token bucket size (also known as the burst) for _relayed traffic_ to the given number of bytes in each direction\&. They do not include directory fetches by the relay (from authority or other relays), because that is considered "client" activity\&. (Default: 0)
.RE
.PP
\fBPerConnBWRate\fR \fIN\fR \fBbytes\fR|\fBKBytes\fR|\fBMBytes\fR|\fBGBytes\fR|\fBTBytes\fR|\fBKBits\fR|\fBMBits\fR|\fBGBits\fR|\fBTBits\fR
.RS 4
If this option is set manually, or via the "perconnbwrate" consensus field, Tor will use it for separate rate limiting for each connection from a non\-relay\&. (Default: 0)
.RE
.PP
\fBPerConnBWBurst\fR \fIN\fR \fBbytes\fR|\fBKBytes\fR|\fBMBytes\fR|\fBGBytes\fR|\fBTBytes\fR|\fBKBits\fR|\fBMBits\fR|\fBGBits\fR|\fBTBits\fR
.RS 4
If this option is set manually, or via the "perconnbwburst" consensus field, Tor will use it for separate rate limiting for each connection from a non\-relay\&. (Default: 0)
.RE
.PP
\fBClientTransportPlugin\fR \fItransport\fR socks4|socks5 \fIIP\fR:\fIPORT\fR, \fBClientTransportPlugin\fR \fItransport\fR exec \fIpath\-to\-binary\fR [options]
.RS 4
In its first form, when set along with a corresponding Bridge line, the Tor client forwards its traffic to a SOCKS\-speaking proxy on "IP:PORT"\&. (IPv4 addresses should written as\-is; IPv6 addresses should be wrapped in square brackets\&.) It\(cqs the duty of that proxy to properly forward the traffic to the bridge\&.

In its second form, when set along with a corresponding Bridge line, the Tor client launches the pluggable transport proxy executable in
\fIpath\-to\-binary\fR
using
\fIoptions\fR
as its command\-line options, and forwards its traffic to it\&. It\(cqs the duty of that proxy to properly forward the traffic to the bridge\&.
.RE
.PP
\fBServerTransportPlugin\fR \fItransport\fR exec \fIpath\-to\-binary\fR [options]
.RS 4
The Tor relay launches the pluggable transport proxy in
\fIpath\-to\-binary\fR
using
\fIoptions\fR
as its command\-line options, and expects to receive proxied client traffic from it\&.
.RE
.PP
\fBServerTransportListenAddr\fR \fItransport\fR \fIIP\fR:\fIPORT\fR
.RS 4
When this option is set, Tor will suggest
\fIIP\fR:\fIPORT\fR
as the listening address of any pluggable transport proxy that tries to launch
\fItransport\fR\&. (IPv4 addresses should written as\-is; IPv6 addresses should be wrapped in square brackets\&.)
.RE
.PP
\fBServerTransportOptions\fR \fItransport\fR \fIk=v\fR \fIk=v\fR \&...
.RS 4
When this option is set, Tor will pass the
\fIk=v\fR
parameters to any pluggable transport proxy that tries to launch
\fItransport\fR\&.

(Example: ServerTransportOptions obfs45 shared\-secret=bridgepasswd cache=/var/lib/tor/cache)
.RE
.PP
\fBExtORPort\fR [\fIaddress\fR:]\fIport\fR|\fBauto\fR
.RS 4
Open this port to listen for Extended ORPort connections from your pluggable transports\&.
.RE
.PP
\fBExtORPortCookieAuthFile\fR \fIPath\fR
.RS 4
If set, this option overrides the default location and file name for the Extended ORPort\(cqs cookie file \(em the cookie file is needed for pluggable transports to communicate through the Extended ORPort\&.
.RE
.PP
\fBExtORPortCookieAuthFileGroupReadable\fR \fB0\fR|\fB1\fR
.RS 4
If this option is set to 0, don\(cqt allow the filesystem group to read the Extended OR Port cookie file\&. If the option is set to 1, make the cookie file readable by the default GID\&. [Making the file readable by other groups is not yet implemented; let us know if you need this for some reason\&.] (Default: 0)
.RE
.PP
\fBConnLimit\fR \fINUM\fR
.RS 4
The minimum number of file descriptors that must be available to the Tor process before it will start\&. Tor will ask the OS for as many file descriptors as the OS will allow (you can find this by "ulimit \-H \-n")\&. If this number is less than ConnLimit, then Tor will refuse to start\&.

You probably don\(cqt need to adjust this\&. It has no effect on Windows since that platform lacks getrlimit()\&. (Default: 1000)
.RE
.PP
\fBDisableNetwork\fR \fB0\fR|\fB1\fR
.RS 4
When this option is set, we don\(cqt listen for or accept any connections other than controller connections, and we close (and don\(cqt reattempt) any outbound connections\&. Controllers sometimes use this option to avoid using the network until Tor is fully configured\&. (Default: 0)
.RE
.PP
\fBConstrainedSockets\fR \fB0\fR|\fB1\fR
.RS 4
If set, Tor will tell the kernel to attempt to shrink the buffers for all sockets to the size specified in
\fBConstrainedSockSize\fR\&. This is useful for virtual servers and other environments where system level TCP buffers may be limited\&. If you\(cqre on a virtual server, and you encounter the "Error creating network socket: No buffer space available" message, you are likely experiencing this problem\&.

The preferred solution is to have the admin increase the buffer pool for the host itself via /proc/sys/net/ipv4/tcp_mem or equivalent facility; this configuration option is a second\-resort\&.

The DirPort option should also not be used if TCP buffers are scarce\&. The cached directory requests consume additional sockets which exacerbates the problem\&.

You should
\fBnot\fR
enable this feature unless you encounter the "no buffer space available" issue\&. Reducing the TCP buffers affects window size for the TCP stream and will reduce throughput in proportion to round trip time on long paths\&. (Default: 0)
.RE
.PP
\fBConstrainedSockSize\fR \fIN\fR \fBbytes\fR|\fBKBytes\fR
.RS 4
When
\fBConstrainedSockets\fR
is enabled the receive and transmit buffers for all sockets will be set to this limit\&. Must be a value between 2048 and 262144, in 1024 byte increments\&. Default of 8192 is recommended\&.
.RE
.PP
\fBControlPort\fR \fIPORT\fR|\fBunix:\fR\fIpath\fR|\fBauto\fR [\fIflags\fR]
.RS 4
If set, Tor will accept connections on this port and allow those connections to control the Tor process using the Tor Control Protocol (described in control\-spec\&.txt in
torspec)\&. Note: unless you also specify one or more of
\fBHashedControlPassword\fR
or
\fBCookieAuthentication\fR, setting this option will cause Tor to allow any process on the local host to control it\&. (Setting both authentication methods means either method is sufficient to authenticate to Tor\&.) This option is required for many Tor controllers; most use the value of 9051\&. If a unix domain socket is used, you may quote the path using standard C escape sequences\&. Set it to "auto" to have Tor pick a port for you\&. (Default: 0)

Recognized flags are\&...
.PP
\fBGroupWritable\fR
.RS 4
Unix domain sockets only: makes the socket get created as group\-writable\&.
.RE
.PP
\fBWorldWritable\fR
.RS 4
Unix domain sockets only: makes the socket get created as world\-writable\&.
.RE
.PP
\fBRelaxDirModeCheck\fR
.RS 4
Unix domain sockets only: Do not insist that the directory that holds the socket be read\-restricted\&.
.RE
.RE
.PP
\fBControlSocket\fR \fIPath\fR
.RS 4
Like ControlPort, but listens on a Unix domain socket, rather than a TCP socket\&.
\fI0\fR
disables ControlSocket (Unix and Unix\-like systems only\&.)
.RE
.PP
\fBControlSocketsGroupWritable\fR \fB0\fR|\fB1\fR
.RS 4
If this option is set to 0, don\(cqt allow the filesystem group to read and write unix sockets (e\&.g\&. ControlSocket)\&. If the option is set to 1, make the control socket readable and writable by the default GID\&. (Default: 0)
.RE
.PP
\fBHashedControlPassword\fR \fIhashed_password\fR
.RS 4
Allow connections on the control port if they present the password whose one\-way hash is
\fIhashed_password\fR\&. You can compute the hash of a password by running "tor \-\-hash\-password
\fIpassword\fR"\&. You can provide several acceptable passwords by using more than one HashedControlPassword line\&.
.RE
.PP
\fBCookieAuthentication\fR \fB0\fR|\fB1\fR
.RS 4
If this option is set to 1, allow connections on the control port when the connecting process knows the contents of a file named "control_auth_cookie", which Tor will create in its data directory\&. This authentication method should only be used on systems with good filesystem security\&. (Default: 0)
.RE
.PP
\fBCookieAuthFile\fR \fIPath\fR
.RS 4
If set, this option overrides the default location and file name for Tor\(cqs cookie file\&. (See CookieAuthentication above\&.)
.RE
.PP
\fBCookieAuthFileGroupReadable\fR \fB0\fR|\fB1\fR
.RS 4
If this option is set to 0, don\(cqt allow the filesystem group to read the cookie file\&. If the option is set to 1, make the cookie file readable by the default GID\&. [Making the file readable by other groups is not yet implemented; let us know if you need this for some reason\&.] (Default: 0)
.RE
.PP
\fBControlPortWriteToFile\fR \fIPath\fR
.RS 4
If set, Tor writes the address and port of any control port it opens to this address\&. Usable by controllers to learn the actual control port when ControlPort is set to "auto"\&.
.RE
.PP
\fBControlPortFileGroupReadable\fR \fB0\fR|\fB1\fR
.RS 4
If this option is set to 0, don\(cqt allow the filesystem group to read the control port file\&. If the option is set to 1, make the control port file readable by the default GID\&. (Default: 0)
.RE
.PP
\fBDataDirectory\fR \fIDIR\fR
.RS 4
Store working data in DIR\&. Can not be changed while tor is running\&. (Default: ~/\&.tor if your home directory is not /; otherwise, /usr/local/var/lib/tor\&. On Windows, the default is your ApplicationData folder\&.)
.RE
.PP
\fBDataDirectoryGroupReadable\fR \fB0\fR|\fB1\fR
.RS 4
If this option is set to 0, don\(cqt allow the filesystem group to read the DataDirectory\&. If the option is set to 1, make the DataDirectory readable by the default GID\&. (Default: 0)
.RE
.PP
\fBCacheDirectory\fR \fIDIR\fR
.RS 4
Store cached directory data in DIR\&. Can not be changed while tor is running\&. (Default: uses the value of DataDirectory\&.)
.RE
.PP
\fBCacheDirectoryGroupReadable\fR \fB0\fR|\fB1\fR
.RS 4
If this option is set to 0, don\(cqt allow the filesystem group to read the CacheDirectory\&. If the option is set to 1, make the CacheDirectory readable by the default GID\&. (Default: 0)
.RE
.PP
\fBFallbackDir\fR \fIipv4address\fR:\fIport\fR orport=\fIport\fR id=\fIfingerprint\fR [weight=\fInum\fR] [ipv6=\fB[\fR\fIipv6address\fR\fB]\fR:\fIorport\fR]
.RS 4
When we\(cqre unable to connect to any directory cache for directory info (usually because we don\(cqt know about any yet) we try a directory authority\&. Clients also simultaneously try a FallbackDir, to avoid hangs on client startup if a directory authority is down\&. Clients retry FallbackDirs more often than directory authorities, to reduce the load on the directory authorities\&. By default, the directory authorities are also FallbackDirs\&. Specifying a FallbackDir replaces Tor\(cqs default hard\-coded FallbackDirs (if any)\&. (See the
\fBDirAuthority\fR
entry for an explanation of each flag\&.)
.RE
.PP
\fBUseDefaultFallbackDirs\fR \fB0\fR|\fB1\fR
.RS 4
Use Tor\(cqs default hard\-coded FallbackDirs (if any)\&. (When a FallbackDir line is present, it replaces the hard\-coded FallbackDirs, regardless of the value of UseDefaultFallbackDirs\&.) (Default: 1)
.RE
.PP
\fBDirAuthority\fR [\fInickname\fR] [\fBflags\fR] \fIipv4address\fR:\fIport\fR \fIfingerprint\fR
.RS 4
Use a nonstandard authoritative directory server at the provided address and port, with the specified key fingerprint\&. This option can be repeated many times, for multiple authoritative directory servers\&. Flags are separated by spaces, and determine what kind of an authority this directory is\&. By default, an authority is not authoritative for any directory style or version unless an appropriate flag is given\&. Tor will use this authority as a bridge authoritative directory if the "bridge" flag is set\&. If a flag "orport=\fBport\fR" is given, Tor will use the given port when opening encrypted tunnels to the dirserver\&. If a flag "weight=\fBnum\fR" is given, then the directory server is chosen randomly with probability proportional to that weight (default 1\&.0)\&. If a flag "v3ident=\fBfp\fR" is given, the dirserver is a v3 directory authority whose v3 long\-term signing key has the fingerprint
\fBfp\fR\&. Lastly, if an "ipv6=\fB[\fR\fIipv6address\fR\fB]\fR:\fIorport\fR" flag is present, then the directory authority is listening for IPv6 connections on the indicated IPv6 address and OR Port\&.

Tor will contact the authority at
\fIipv4address\fR
to download directory documents\&. The provided
\fIport\fR
value is a dirport; clients ignore this in favor of the specified "orport=" value\&. If an IPv6 ORPort is supplied, Tor will also download directory documents at the IPv6 ORPort\&.

If no
\fBDirAuthority\fR
line is given, Tor will use the default directory authorities\&. NOTE: this option is intended for setting up a private Tor network with its own directory authorities\&. If you use it, you will be distinguishable from other users, because you won\(cqt believe the same authorities they do\&.
.RE
.PP
\fBDirAuthorityFallbackRate\fR \fINUM\fR
.RS 4
When configured to use both directory authorities and fallback directories, the directory authorities also work as fallbacks\&. They are chosen with their regular weights, multiplied by this number, which should be 1\&.0 or less\&. The default is less than 1, to reduce load on authorities\&. (Default: 0\&.1)
.RE
.sp
\fBAlternateDirAuthority\fR [\fInickname\fR] [\fBflags\fR] \fIipv4address\fR:\fIport\fR \fIfingerprint\fR
.PP
\fBAlternateBridgeAuthority\fR [\fInickname\fR] [\fBflags\fR] \fIipv4address\fR:\fIport\fR \fI fingerprint\fR
.RS 4
These options behave as DirAuthority, but they replace fewer of the default directory authorities\&. Using AlternateDirAuthority replaces the default Tor directory authorities, but leaves the default bridge authorities in place\&. Similarly, AlternateBridgeAuthority replaces the default bridge authority, but leaves the directory authorities alone\&.
.RE
.PP
\fBDisableAllSwap\fR \fB0\fR|\fB1\fR
.RS 4
If set to 1, Tor will attempt to lock all current and future memory pages, so that memory cannot be paged out\&. Windows, OS X and Solaris are currently not supported\&. We believe that this feature works on modern Gnu/Linux distributions, and that it should work on *BSD systems (untested)\&. This option requires that you start your Tor as root, and you should use the
\fBUser\fR
option to properly reduce Tor\(cqs privileges\&. Can not be changed while tor is running\&. (Default: 0)
.RE
.PP
\fBDisableDebuggerAttachment\fR \fB0\fR|\fB1\fR
.RS 4
If set to 1, Tor will attempt to prevent basic debugging attachment attempts by other processes\&. This may also keep Tor from generating core files if it crashes\&. It has no impact for users who wish to attach if they have CAP_SYS_PTRACE or if they are root\&. We believe that this feature works on modern Gnu/Linux distributions, and that it may also work on *BSD systems (untested)\&. Some modern Gnu/Linux systems such as Ubuntu have the kernel\&.yama\&.ptrace_scope sysctl and by default enable it as an attempt to limit the PTRACE scope for all user processes by default\&. This feature will attempt to limit the PTRACE scope for Tor specifically \- it will not attempt to alter the system wide ptrace scope as it may not even exist\&. If you wish to attach to Tor with a debugger such as gdb or strace you will want to set this to 0 for the duration of your debugging\&. Normal users should leave it on\&. Disabling this option while Tor is running is prohibited\&. (Default: 1)
.RE
.PP
\fBFetchDirInfoEarly\fR \fB0\fR|\fB1\fR
.RS 4
If set to 1, Tor will always fetch directory information like other directory caches, even if you don\(cqt meet the normal criteria for fetching early\&. Normal users should leave it off\&. (Default: 0)
.RE
.PP
\fBFetchDirInfoExtraEarly\fR \fB0\fR|\fB1\fR
.RS 4
If set to 1, Tor will fetch directory information before other directory caches\&. It will attempt to download directory information closer to the start of the consensus period\&. Normal users should leave it off\&. (Default: 0)
.RE
.PP
\fBFetchHidServDescriptors\fR \fB0\fR|\fB1\fR
.RS 4
If set to 0, Tor will never fetch any hidden service descriptors from the rendezvous directories\&. This option is only useful if you\(cqre using a Tor controller that handles hidden service fetches for you\&. (Default: 1)
.RE
.PP
\fBFetchServerDescriptors\fR \fB0\fR|\fB1\fR
.RS 4
If set to 0, Tor will never fetch any network status summaries or server descriptors from the directory servers\&. This option is only useful if you\(cqre using a Tor controller that handles directory fetches for you\&. (Default: 1)
.RE
.PP
\fBFetchUselessDescriptors\fR \fB0\fR|\fB1\fR
.RS 4
If set to 1, Tor will fetch every consensus flavor, and all server descriptors and authority certificates referenced by those consensuses, except for extra info descriptors\&. When this option is 1, Tor will also keep fetching descriptors, even when idle\&. If set to 0, Tor will avoid fetching useless descriptors: flavors that it is not using to build circuits, and authority certificates it does not trust\&. When Tor hasn\(cqt built any application circuits, it will go idle, and stop fetching descriptors\&. This option is useful if you\(cqre using a tor client with an external parser that uses a full consensus\&. This option fetches all documents except extrainfo descriptors,
\fBDirCache\fR
fetches and serves all documents except extrainfo descriptors,
\fBDownloadExtraInfo\fR* fetches extrainfo documents, and serves them if
\fBDirCache\fR
is on, and
\fBUseMicrodescriptors\fR
changes the flavour of consensues and descriptors that is fetched and used for building circuits\&. (Default: 0)
.RE
.PP
\fBHTTPProxy\fR \fIhost\fR[:\fIport\fR]
.RS 4
Tor will make all its directory requests through this host:port (or host:80 if port is not specified), rather than connecting directly to any directory servers\&. (DEPRECATED: As of 0\&.3\&.1\&.0\-alpha you should use HTTPSProxy\&.)
.RE
.PP
\fBHTTPProxyAuthenticator\fR \fIusername:password\fR
.RS 4
If defined, Tor will use this username:password for Basic HTTP proxy authentication, as in RFC 2617\&. This is currently the only form of HTTP proxy authentication that Tor supports; feel free to submit a patch if you want it to support others\&. (DEPRECATED: As of 0\&.3\&.1\&.0\-alpha you should use HTTPSProxyAuthenticator\&.)
.RE
.PP
\fBHTTPSProxy\fR \fIhost\fR[:\fIport\fR]
.RS 4
Tor will make all its OR (SSL) connections through this host:port (or host:443 if port is not specified), via HTTP CONNECT rather than connecting directly to servers\&. You may want to set
\fBFascistFirewall\fR
to restrict the set of ports you might try to connect to, if your HTTPS proxy only allows connecting to certain ports\&.
.RE
.PP
\fBHTTPSProxyAuthenticator\fR \fIusername:password\fR
.RS 4
If defined, Tor will use this username:password for Basic HTTPS proxy authentication, as in RFC 2617\&. This is currently the only form of HTTPS proxy authentication that Tor supports; feel free to submit a patch if you want it to support others\&.
.RE
.PP
\fBSandbox\fR \fB0\fR|\fB1\fR
.RS 4
If set to 1, Tor will run securely through the use of a syscall sandbox\&. Otherwise the sandbox will be disabled\&. The option is currently an experimental feature\&. It only works on Linux\-based operating systems, and only when Tor has been built with the libseccomp library\&. This option can not be changed while tor is running\&.

When the Sandbox is 1, the following options can not be changed when tor is running: Address ConnLimit CookieAuthFile DirPortFrontPage ExtORPortCookieAuthFile Logs ServerDNSResolvConfFile Tor must remain in client or server mode (some changes to ClientOnly and ORPort are not allowed)\&. (Default: 0)
.RE
.PP
\fBSocks4Proxy\fR \fIhost\fR[:\fIport\fR]
.RS 4
Tor will make all OR connections through the SOCKS 4 proxy at host:port (or host:1080 if port is not specified)\&.
.RE
.PP
\fBSocks5Proxy\fR \fIhost\fR[:\fIport\fR]
.RS 4

Public/Final/tor-0.3.3.9/etc/tor/tor.sh
###########################################################################
##  Copyright (C) Wizardry and Steamworks 2013 - License: GNU GPLv3      ##
##  Please see: http://www.gnu.org/licenses/gpl.html for legal details,  ##
##  rights of fair usage, the disclaimer and warranty conditions.        ##
###########################################################################


# MCApollo's compiled, WASs launchdaemon and Settings Panel:
# Changes: This runs as root, so drop privs when running commands. 'su mobile' is kinda pointless, writing this now.
# Changes: Edit names to avoid conflict with the BigBoss Package.
# com.mc.tor.plist is chmod 600 and owned by mobile, can't be read as nobody.

STATUS=`ps -ax | grep '[t]or' | grep 'hush' | awk '{ print $1 }'`
SETTING=$(su mobile -s /bin/sh -c "plutil -key torEnabled /private/var/mobile/Library/Preferences/com.mc.tor.plist")

if [ $SETTING -eq 0 ]; then
  kill -s TERM $STATUS
else
   cd /tmp # remember that nobody REALLY has no perms or write access without giving it to him.
   su nobody -s /bin/sh -c "/usr/local/bin/tor --defaults-torrc --hush"
fi
